---
title: "Understanding The Common Mechanism And A Suggestion For Improvement"
author: "Phil Palmer"
date: "31st July 2023"
output:
  bookdown::pdf_document2:
    toc: true
    number_sections: true
    fig_caption: true
    includes:
      in_header: preamble.tex
#   pdf_document: default
#   beamer_presentation:
#     latex_engine: pdflatex
#     highlight: haddock
#     fig_width: 7 
#     fig_height: 3
#     includes:
#         in_header: preamble.tex
bibliography: references/references.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(dplyr)
library(ggplot2)
library(knitr)
library(rmarkdown)
```

<!--
```{bash, include=FALSE, echo=FALSE}
# Activate the conda environment
source ~/anaconda3/etc/profile.d/conda.sh
conda activate commec-dev
```
-->

\hfill \break
\hfill \break
Code for this handout is available at: https://github.com/PhilPalmer/about-common-mechanism
\clearpage
\newpage

# Introduction

## About me

I'm a final year biology PhD student at the University of Cambridge, working on computational methods to design broad-spectrum vaccines and antibodies.
As shown in the image below, I completed my undergraduate degree in biology at the University of Southampton,
worked for two years in bioinformatics at a biotech start-up in London,
and have interned at the Massachusetts Institute of Technology (MIT) working in metagenomics.

```{r, echo=FALSE, out.width="65%"}
knitr::include_graphics("images/career_path.png")
```

# Background

## DNA synthesis

Biotechnology is rapidly becoming more capable, accessible and affordable [@gerstein_emerging_2024].
For example, as shown in the figure below, the cost of DNA sequencing and gene synthesis has decreased by several orders of magnitude over the past two decades [@carlson_dna_2022].


```{r, echo=FALSE, eval=FALSE, out.width="60%", fig.cap="Price Per Base of DNA Sequencing and Synthesis (Data from Carlson 2022)"}
library(ggplot2)
library(RColorBrewer)
library(tidyr)

data <- read.csv("data/dna_synthesis_cost.csv")

# Reshape the data for plotting
data_long <- pivot_longer(data, cols = -Year, names_to = "Type", values_to = "Price")

# Fix legend names by replacing periods with spaces
data_long$Type <- gsub("\\.", " ", data_long$Type)
data_long$Type <- gsub(" Column", "(Column)", data_long$Type)

# Remove NA values
data_long <- data_long %>% filter(!is.na(Price))

# Create the plot
ggplot(data_long, aes(x = Year, y = Price, color = Type, shape = Type)) +
  geom_line(na.rm = TRUE) +
  geom_point() +
  scale_y_log10(breaks = scales::trans_breaks("log10", function(x) 10^x),
                labels = scales::trans_format("log10", scales::math_format(10^.x))) +
  scale_x_continuous(breaks = seq(1990, 2020, 2)) +
  labs(title = "",
       x = "",
       y = "US Dollars") +
  scale_color_brewer(palette = "Set1") +
  theme_minimal() +
  theme(legend.position = "bottom", 
        legend.title = element_blank(),
        panel.grid.major.y = element_line(color = "grey80"),
        panel.grid.minor.y = element_line(color = "grey90"),
        panel.grid.major.x = element_blank())

# Save the plot
ggsave("images/price_dna_sequencing_synthesis.png", dpi = 300)
```

\begin{minipage}{0.65\textwidth}
\centering
\includegraphics[width=\linewidth]{images/price_dna_sequencing_synthesis.png}
\captionof{figure}{Price Per Base of DNA Sequencing and Synthesis (Data from Carlson 2022).}
\end{minipage}
\hfill
\begin{minipage}{0.32\textwidth}
This rapid decrease in the cost of DNA synthesis is due to the modern chemical synthesis of DNA which uses phosphoramidite chemistry.
This method was original developed in the 1980s and has been refined since then to rapidly and accurately synthesise DNA.
The process involves protecting the reactive 5'-hydroxyl group of each nucleotide with a dimethoxytrityl (DMT) group.
During synthesis, this DMT group is selectively removed, allowing for the controlled, sequential addition of nucleotides to build the desired DNA sequence (Frank 2024).
\end{minipage}

## Misuse risk

The rapid decrease in DNA synthesis costs has driven progress in biotechnology, helping to address global challenges in health, climate change and food security [@wheeler_developing_2024]. However, DNA synthesis, combined with other dual-use technologies, poses misuse risks by malicious actors:

1. **Reverse genetics protocols:** The 2018 synthesis of horsepox virus raised concerns about potential smallpox virus synthesis [@noyce_construction_2018]. Reverse genetics protocols are now available to recreate many pathogens from their DNA sequence, as shown by SARS-CoV-2 synthesis [@thi_nhu_thao_rapid_2020].

2. **Gain-of-function research of concern (GOFRC):** GOFRC studies have identified pathogens with pandemic potential, such as H5N1 influenza, that could be synthesised and released by malicious actors [@herfst_airborne_2012].

3. **AI-enabled biological design tools:** Advances in deep learning for protein design could enhance pathogens' pandemic potential, e.g., by conferring immune evasion capabilities [@thadani_learning_2023].

While rare, there is some historical precedent for bioterrorism. Perhaps the two most notable incidents are the Aum Shinrikyo cult's attempts to synthesise and release anthrax and botulinum toxin in 1990s Japan, and the 2001 US anthrax attacks [@sugishima_aum_2003]. However, these incidents used outdated biotechnologies, and the bioterrorism risk has since evolved.

## DNA synthesis screening

Screening DNA synthesis orders to prevent harmful or dangerous sequence synthesis is key to mitigating misuse risks.
As of 2022, at least 57 DNA synthesis companies existed globally, with nearly half being International Gene Synthesis Consortium (IGSC) members, performing voluntary DNA sequence screening, as shown in Figure \ref{fig:dna_synthesis_companies} [@delaney_airtable_2022].^[A common statistic suggests 80% of DNA synthesis orders are screened. While hard evidence for this figure is lacking, it seems reasonable given that the IGSC represents the 10 largest DNA synthesis companies [@williams_preventing_2023].]
Additionally, there are several software methods available to DNA synthesis providers to screen DNA sequences for pathogens and toxins, as shown in Table \ref{tab:screening_methods} [@frank_evaluation_2024].


```{r, echo=FALSE, eval=FALSE, out.width="70%", fig.cap="Number of DNA Synthesis Companies per Country (Data from Delaney and Pálya 2022)"}
library(ggplot2)
library(dplyr)
library(RColorBrewer)

# Read the CSV file
dna_synthesis_companies_csv <- "data/dna_synthesis_companies.csv"
dna_companies <- read.csv(dna_synthesis_companies_csv)

# Get the % of companies that are members of the IGSC (where "screening" == "IGSC")
igsc_membership <- dna_companies %>%
    count(Screening) %>%
    mutate(percentage = n / sum(n) * 100)
print(igsc_membership)

# Convert the Screening column to a binary variable indicating whether they screen or not
# dna_companies <- dna_companies %>%
#   mutate(Screening = ifelse(Screening == "No", "No", "Yes"))

# # Summarize the number of companies per country and screening status
# country_summary <- dna_companies %>%
#   group_by(Country, Screening) %>%
#   summarise(CompanyCount = n(), .groups = 'drop') %>%
#   arrange(desc(CompanyCount))

# # Plot the number of companies per country with screening status
# ggplot(country_summary, aes(x = reorder(Country, -CompanyCount), y = CompanyCount, fill = Screening)) +
#   geom_bar(stat = 'identity') +
#   coord_flip() +
#   labs(title = '',
#        x = 'Country',
#        y = 'Number of Companies',
#        fill = 'Perform Screening') +
#   scale_fill_brewer(palette = "Set1") +
#   theme_minimal() +
#   theme(legend.position = "bottom")

# # Save the plot
# ggsave("images/dna_synthesis_companies.png")
```

\begin{minipage}{0.47\textwidth}
\centering
\includegraphics[width=\linewidth]{images/dna_synthesis_companies.png}
\captionof{figure}{Number of DNA Synthesis Companies per Country (Data from Delaney and Pálya 2022).}
\label{fig:dna_synthesis_companies}
\end{minipage}
\hfill
\begin{minipage}{0.5\textwidth}
\centering
\captionof{table}{Software methods available for DNA sequence screening.}
\label{tab:screening_methods}
\begin{tabular}{p{0.3\linewidth} p{0.65\linewidth}}
\toprule
\textbf{Method} & \textbf{Description} \\
\midrule
PathoFact & Identifies virulence factors and antibiotic resistance in metagenomic datasets. \\
DeePac and PaPrBaG & Uses ML to predict pathogenicity from bacterial genomes. \\
SeqScreen & Screens DNA sequence orders for pathogen sequences using ML. \\
SecureDNA & Screens orders against a secure database while protecting client query privacy. \\
ThreatSeq & Uses a curated blacklist and predictive model to screen for biothreats. \\
FastNA & Detects harmful sequences with diagnostic signature generation. \\
BLiSS & Aligns sequences to databases to identify sequences of concern. \\
\bottomrule
\end{tabular}
\end{minipage}


# The International Common Mechanism for DNA Synthesis Screening

## Overview

As shown in Figure \ref{fig:common_mechanism_overview}, the International Common Mechanism will provide resources for each step of synthesis screening, including software for sequence screening and resources to facilitate customer screening.

\begin{figure}[ht]
\centering
\includegraphics[width=0.85\linewidth]{images/common_mechanism_overview.png}
\captionof{figure}{Steps in the DNA synthesis screening process (from IBBIS 2024).}
\label{fig:common_mechanism_overview}
\end{figure}

## Sequence screening

The Common Mechanism's software for sequence screening was recently made available on GitHub for beta testing.
The software aims to establish a baseline for DNA synthesis screening, addressing three key challenges:

1. Reducing costs and meeting commercial needs for synthesis providers
    - Cheaper screening processes are necessary as synthesis costs decrease and order volumes increase, creating an economic burden for providers.
    - Providers prefer in-house screening to protect sensitive customer data and maintain trust.
2. Building international trust in screening practices
    - There is limited international collaboration on screening standards, with few shared practices.
    - U.S. resources for screening are tied to national defense and not widely accessible or trusted globally.
3. Integrating screening capabilities into benchtop devices
    - As benchtop synthesis devices become more common, integrating effective and efficient screening into these devices is necessary.
    - Many manufacturers plan to use a "phone home" approach for cloud-based screening, raising similar challenges to traditional providers.

The Common Mechanism screens sequences $\geq$ 50 nucleotides and flags sequences of concern using three analysis modules:

\begin{minipage}{0.3\textwidth}
1. \textbf{Biorisk Database Comparison (M1)}: Uses hidden Markov Models models to detect variants using an initial database from public sources limited to sequences in regulated pathogens and toxins.

2. \textbf{Taxonomic Best Match (M2)}: Compares orders against public DNA/protein sequences and cross-references with international control lists using BLAST and/or DIAMOND.

3. \textbf{Benign Gene Identification (M3)}: Evaluates sequences matching regulated, non-viral organisms and identifies known benign functions.
\end{minipage}
\hfill
\begin{minipage}{0.65\textwidth}
\centering
\includegraphics[width=\linewidth]{images/common_mechanism_seq_screening.jpg}
\captionof{figure}{Common Mechanism A) analysis module and B) decision-making framework (from Wheeler, 2024)}
\label{fig:common_mechanism_seq_screening}
\end{minipage}

Figure \ref{fig:common_mechanism_seq_screening}B shows the decision-making framework for the Common Mechanism, which flags all regulated pathogen and toxins, as well as virulence factors from non-viral regulated pathogens consistent with export controls.

## Testing

To evaluate the Common Mechanism, I installed the software on my laptop and tested it using the SARS-CoV-2 Nucleoprotein sequence, which I had previously ordered from a DNA synthesis company as part of my PhD research.
I ran the software in `--fast` mode, which only requires the smaller biorisk and benign databases, rather than the full NCBI databases that demand 275 - 650 GB of disk space.

The screening process successfully identified sequence similarity to the SARS-realted Nucleoprotein:

```{bash, eval=FALSE, echo=TRUE}
commec screen -d dbs/ -o ./fast-cov example_data/cov.fasta --fast
```

```{bash, eval=TRUE, echo=FALSE}
cat ../common-mechanism/fast-cov.screen
```

This was then flagged as a virulence factor:

```{bash, eval=FALSE, echo=TRUE}
commec flag .
```

```{bash, eval=FALSE, echo=FALSE}
cat ../common-mechanism/flags*.csv
```

| query | biorisk | vf | reg_virus | reg_bacteria | reg_euk | mix_reg_non_reg | benign |
|-------|---------|----|-----------|--------------|---------|-----------------|---------
| SARS-2_N | P | F | - | - | - | - | - |

\newpage

# Improving the Common Mechanism

## Expanding the Common Mechanism

Wheeler et al. 2024 suggested three areas for improving the Common Mechanism:

\begin{table}[ht]
\centering
\begin{tabular}{p{0.25\linewidth} p{0.75\linewidth}}
\toprule
\textbf{Suggestion} & \textbf{Example(s)} \\
\midrule
1. Expand beyond regulated pathogens and toxins & Expand the list of sequences of concern to include those not derived from regulated pathogens and toxins, as stated in the updated HHS Screening Framework. \\
2. Update screening as science and policy develop & IBBIS could host and update sequence databases as new sequences of concern are identified or policies change. \\
 & Respond to advances in deep learning methods that could evade screening or better identify sequences of concern. \\
3. Limit information hazards from shared sequence databases & Consistent with the HHS Screening Framework, only share databases containing novel sequences of concern with trusted partners. \\
\bottomrule
\end{tabular}
\caption{Suggestions for improving the Common Mechanism (Wheeler et al. 2024).}
\end{table}

## My Suggestion for Improvement

Building upon the second suggestion, I propose **developing and incorporating an additional analysis module into the Common Mechanism to detect AI-generated sequences.**

### Background

AI models, such as those listed in Table \ref{tab:ai_models}, are increasingly used to design functional proteins with low sequence similarity to naturally occurring proteins [@wheeler_developing_2024].
This presents a challenge for DNA synthesis screening, as AI-generated sequences may evade existing screening methods.

\begin{table}[ht]
\centering
\begin{tabular}{lll}
\toprule
\textbf{Model} & \textbf{Architecture} & \textbf{Modalities} \\
\midrule
ESM-3 & Transformer & Sequence, structure, function \\
RF Diffusion & Diffusion & Sequence, structure \\
EvoDiff & Diffusion & Sequence \\
\bottomrule
\end{tabular}
\caption{Examples of widely used AI models for protein design.}
\label{tab:ai_models}
\end{table}

The detection of AI-generated protein sequences, remains a nascent field with limited direct research available.
Currently, we do not know if AI-generated sequences can be detected reliably.
Here, I propose that methods from other domains, such as large language model (LLM) text detection, could be applied to this problem and the resulting method could be integrated into the Common Mechanism if successful.

### Methods

Detecting AI-generated sequences is essentially a binary classification task.
Perhaps the most reliable current method for detecting AI-generated text is watermarking, where a unique signature is embedded in the generated text during the generation process [@kumarage_survey_2024].
However, as this requires cooperation from AI developers, which may not always be feasible, I propose evaluating post-hoc detection methods based on the following two approaches:

1. Feature-based approaches:
    <!--
    style, semantics and context 
    -->
    - Use a large dataset of labelled AI-generated and naturally occurring proteins to train a classifier to distinguish between the two.
    - Learn features such as amino acid preferences, sequence motifs, or structural characteristics.
    - Evaluate traditional machine learning algorithms and deep learning models, adapting techniques like TriFuseNet for protein sequences.

2. LLM probability function approaches:
    - Employ techniques like DetectGPT by introducing small changes to protein sequences and compare log probabilities using a pre-trained protein language model.
    - This allows for the detection of AI-generated sequences without extensive training data by directly evaluating probability scores of original and perturbed sequences.
   <!-- - Leverage cues from LLM probability functions, such as frequency bias
   - Similar to DetectGPT, predict the likelihood 
   - Employ techniques like DetectGPT, which examines curvatures in the LLM's log probability function.
   - Utilise deep learning methods to mask tokens (amino acids) and predict their likelihood across the sequence.
   - Employ techniques like DetectGPT, 
   -->

<!--
1. Feature-based approaches:
   - Stylometry analysis
   - Structural examination
   - Sequence characteristics

2. LLM probability function:
   - Utilise deep learning methods, such as LLMs, to mask tokens (amino acids) and predict their likelihood across the sequence
   - Fine-tune existing LLMs for this specific task
-->

\begin{minipage}{0.5\textwidth}
However, many challenges must be addressed:

\begin{enumerate}
    \item \textbf{Feasibility:} Determining the extent to which AI-generated and natural sequences can be distinguished (see Figure \ref{fig:esm3_sequences_umap}).
    \item \textbf{Generalisability:} Ensuring the method works across different AI models
    \item \textbf{Accuracy:} Minimising false positives and negatives
    \item \textbf{Adaptability:} Keeping pace with new AI models
    \item \textbf{Computational demands:} Managing the high computational cost of LLM inference and running this additional module
\end{enumerate}
\end{minipage}
\hfill
\begin{minipage}{0.47\textwidth}
\centering
\includegraphics[width=0.8\linewidth]{images/esm3_sequences_umap.png}
\captionof{figure}{UMAP plot of ESM-3 generated sequences (colored by sequence identity) and natural sequences (gray) (Figure from Hayes et al. 2024).}
\label{fig:esm3_sequences_umap}
\end{minipage}

As shown in Figure \ref{fig:common_mechanism_ai_detection}, the AI detection module could be an additional step after the pathogen detection.
This may help prevent AI-dependent evasion of screening, as sequences flagged as AI-generated would be subject to additional scrutiny within the decision-making framework and/or manual investigation.

\begin{figure}[ht]
\centering
\includegraphics[width=0.8\textwidth]{images/common_mechanism_ai.png}
\caption{Integration of AI detection module into the Common Mechanism (adapted from IBBIS 2024).}
\label{fig:common_mechanism_ai_detection}
\end{figure}

### Discussion

Reliable detection of AI-generated protein sequences would be valuable not only for DNA synthesis screening but also for other biosecurity applications, such as early pathogen detection and genetic engineering detection.
By addressing these challenges and integrating AI detection into the Common Mechanism, we can enhance our ability to identify potential biosecurity risks and stay ahead of emerging threats in synthetic biology.

\newpage

<!--
## My suggestion for improvement

For my suggestion for improvement, I would like to build on the second suggestion from Wheeler et al. 2024, of focusing on responding to advances in deep learning methods to improve identification of sequences of concern.
Specefically, I propose that the Common Mechanism should include an additional analysis module that detects if a sequence is AI generated.

Applications
- Novel problem/nascent field but with methods from other fields e.g. ML than can be applied
- DNA synthesis screening e.g. detecting sequences that may have been generated by AI to evade screening
- Useful for other biosecurity applications, e.g. early detection of pathogens and genetic engineering detection

Background
- Methods for protein design using AI
    - Transformers-based models (LLMs) e.g. ESM-3
    - Diffusion models e.g. RF Diffusion (uses structural data) and EvoDiff (uses sequence data only)

Methods
- Binary classification task
- Watermarking vs Post-hoc detection (https://arxiv.org/pdf/2403.01152)
- Post-hoc detection
    - Supervised detectors: Classifier trained on a dataset of AI-generated sequences and non-AI generated sequences
        - Generate dataset
- Two approaches to Post-hoc detection
    1. Features
        - Stylometry
        - Structural
        - Sequence
    2. LLM probability function: - Use deep learning methods, e.g. LLMs model themselves mask tokens (amino acids) and predict the likelihood of the masked token across the sequence
        - Fine-tune existing LLMs for this task
- Look to methods in genetic engineering detection, similar to problem of pathogen detection

Integration into the Common Mechanism
- Include as an additional check after the benign gene identification module
- If flagged as AI-generated, the sequence should be flagged as a sequence of concern
- Would also be useful for manual investigation of sequences

Challenges
- Feasibility of this approach - to what extent AI-generated sequences different from naturally occurring sequences?
    - Can you generalise across different models?
- Keep up to date with advances in AI
- Potential for false positives and negatives
- High computational cost of LLM inference
- Computational cost of running this module

## My suggestion for improvement

While the above suggestions are crucial for reducing misuse risks associated with DNA synthesis, I propose an additional, more fundamental, area for improvement.

\begin{minipage}{0.5\textwidth}
To illustrate, consider DNA synthesis screening as fences preventing access to a hazardous area.
Different companies own parts of the land surrounding this area, and many, including IGSC members, have erected fences.
However, gaps exist where some companies perform no screening. As non-IGSC companies are known, these gaps are visible to potential bad actors.
\end{minipage}
\hfill
\begin{minipage}{0.45\textwidth}
\centering
\includegraphics[width=\linewidth]{images/screening_fence_analogy.png}
\captionof{figure}{DNA synthesis screening analogy (from Esvelt, 2023)}
\end{minipage}

Using this analogy, previous suggestions to expand the Common Mechanism may be likened to building higher fences. However, aligning with the overarching goal of providing a baseline for DNA synthesis screening, I suggest first ensuring all companies have a fence.
In other words, we should **focus on increasing the adoption of the Common Mechanism by DNA synthesis companies that do not currently perform screening**.

Strategies to achieve this could include:

1. **Awareness**: Identify companies not performing screening (there are 31 such companies according to Delaney and Pálya 2022) and conduct targeted outreach about the Common Mechanism and its benefits.
2. **Documentation**: Refine and improve implementation guides to facilitate easier adoption.
3. **Training**: Provide workshops on implementing the Common Mechanism.
4. **Support**: Offer technical assistance to companies during implementation.
5. **Onboarding**: Streamline the process for new companies to adopt the Common Mechanism.
6. **Incentives**: Provide motivation for adoption through financial incentives, recognition, or regulation.

I am not aware of the existing work in this area but by focusing on widespread adoption, we can establish a more comprehensive baseline for DNA synthesis screening, significantly reducing potential misuse risks while building on the valuable suggestions for expansion proposed by Wheeler et al.

-->

# References
